### PR TITLE
Tests for email collaborators pagination

### PR DESCRIPTION
**Pull request by nickvergessen**
Followup to #8596

### CODE REVIEW

File: lib/private/Collaboration/Collaborators/MailPlugin.php [hunk 1]
```diff
@@ -196,7 +200,7 @@ public function search($search, $limit, $offset, ISearchResult $searchResult) {
 		}
 		$searchResult->addResultSet($emailType, $result['wide'], $result['exact']);
 
-		return true;
```
Comment (nickvergessen) [inline lib/private/Collaboration/Collaborators/MailPlugin.php]: this was a pretty bad assumption, so I had to fix all tests and wrote some where it is actually supposed to be true

Comment (blizzz) [reply lib/private/Collaboration/Collaborators/MailPlugin.php]: Its the default for anything that cannot (or could not) paginate. I took it over as it was when i refactored it.

**Comment by blizzz** [comment]
no test case where you look for $reachedEnd = true?

File: lib/private/Collaboration/Collaborators/MailPlugin.php [hunk 2]
```diff
@@ -172,10 +172,14 @@ public function search($search, $limit, $offset, ISearchResult $searchResult) {
 			}
 		}
 
+		$reachedEnd = true;
 		if (!$this->shareeEnumeration) {
 			$result['wide'] = [];
 			$userResults['wide'] = [];
 		} else {
+			$reachedEnd = (count($result['wide']) < $offset + $limit) ||
```
Comment (MorrisJobke) [inline lib/private/Collaboration/Collaborators/MailPlugin.php]: Isn't the end reached when both sets don't have further elements?

Comment (nickvergessen) [reply lib/private/Collaboration/Collaborators/MailPlugin.php]: yeah should have been `&&`

also the answer for @blizzz  comment. I messed up when inverting the logic from "more results" to "end reached"

Comment (nickvergessen) [reply lib/private/Collaboration/Collaborators/MailPlugin.php]: fixed

**Comment by blizzz** [comment]
failing tests

**Comment by nickvergessen** [comment]
I messed up conflict resolution on rebase.
Fixed now

**Review by rullzer — APPROVED** [review]

**Comment by nickvergessen** [comment]
Backport in https://github.com/nextcloud/server/pull/8749

**Review by blizzz — APPROVED** [review]

**Pull request merged**